### DISCUSSION TITLE
How to force utf-8 encoding on WebView2?

### DISCUSSION THREAD

**Discussion by HTprogrammer** [MicrosoftEdge/WebView2Feedback#3981] [Q&A]
I am using webVew2 with C# VS2022 environment. 
On accessing url, server responses error message in Japanese encoded utf-8.
But WebView2 automatically treats as Shit-JIS. On the sceen of WebView2, un-readable Japanese characters are shown. 
I would like to correct this, to force utf-8 encoding on WebView2. Is it possible?
Below are my environment
WebView2: 1.0.1340
Windows Pro 21H2
Visual Studio 2022 C#

**Comment by LiangTheDev** [comment]
If the error message is displayed by WebView as un-readable Japanese, it suggests that there are some OS level settings causing the WebView code to think that the default encoding is JIS and using that encoding, when there is no `Content-Type` header from server or related `meta` tag on the page. When this happens, it is likely that we will get similar behavior in browser.
Is it possible to ensure that the server sets `Content-Type` header like like `Content-Type: text/html; charset=UTF-8` when sending back the error message?
